external/avro/src/main/scala/org/apache/spark/sql/avro/functions.scala

```diff
@@ -28,39 +28,59 @@ object functions {
 // scalastyle:on: object.name
 
   /**
-   * Converts a binary column of avro format into its corresponding catalyst value. The specified
-   * schema must match the read data, otherwise the behavior is undefined: it may fail or return
-   * arbitrary result.
+   * Converts a binary column of avro format into its corresponding catalyst value. If a writer's
+   * schema is provided, a different (but compatible) schema can be used for reading. If no writer's
+   * schema is provided, the specified schema must match the read data, otherwise the behavior is
+   * undefined: it may fail or return arbitrary result.
    *
    * @param data the binary column.
    * @param jsonFormatSchema the avro schema in JSON string format.
+   * @param writerJsonFormatSchema the avro schema in JSON string format used to serialize the data.
    *
    * @since 3.0.0
    */
   @Experimental
   def from_avro(
       data: Column,
-      jsonFormatSchema: String): Column = {
-    new Column(AvroDataToCatalyst(data.expr, jsonFormatSchema, Map.empty))
+      jsonFormatSchema: String,
+      writerJsonFormatSchema: Option[String]): Column = {
```

Comment:
shall we have it with default value to `None`?

Reply:
it's not possible because the other version already has a default value: 
`[error] in object functions, multiple overloaded alternatives of method from_avro define default arguments`